Kudu.Core/Jobs/ContinuousJobRunner.cs

```diff
@@ -297,5 +302,26 @@ protected virtual void Dispose(bool disposing)
                 }
             }
         }
+
+        private int GetAvailablePortForWebJob()
+        {
+            if (JobSettings.IsWebHook)
+            {
+                var listener = new TcpListener(IPAddress.Loopback, 0);
+                try
+                {
+                    listener.Start();
+                    return ((IPEndPoint)listener.LocalEndpoint).Port;
+                }
+                finally
+                {
+                    listener.Stop();
```

Comment:
If it's sequential such that 2 calls won't get the same even if neither actually uses the port, then we're fine.

In any case, we have a fallback if some extremely unlikely scenario causes a conflict: the job will fail to start when it tries to listen on the port, and we'll restart it.

So I think we're fine.